tests/accounts.rs

```diff
@@ -41,6 +41,104 @@ async fn account_ordering() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn remove_latest_account() -> Result<()> {
+    std::fs::remove_dir_all("test-storage/remove_latest_account").unwrap_or(());
+
+    let client_options = ClientOptions::new()
+        .with_node("http://localhost:14265")?
+        .with_node_sync_disabled();
+
+    let (third_account_index, forth_account_index) = {
+        // Mnemonic without balance.
+        let secret_manager = MnemonicSecretManager::try_from_mnemonic(
+            "inhale gorilla deny three celery song category owner lottery rent author wealth penalty crawl hobby obtain glad warm early rain clutch slab august bleak",
+        )?;
+
+        let manager = AccountManager::builder()
+            .with_secret_manager(SecretManager::Mnemonic(secret_manager))
+            .with_client_options(client_options.clone())
+            .with_storage_path("test-storage/remove_latest_account")
+            .finish()
+            .await?;
+
+        // Create two accounts.
+        let first_account = manager.create_account().finish().await?;
+        let _second_account = manager.create_account().finish().await?;
+        assert!(manager.get_accounts().await.unwrap().len() == 2);
+
+        // Remove `second_account`.
+        let _ = manager
+            .remove_latest_account()
+            .await
+            .expect("cannot remove latest account");
+
+        // Check if the `second_account` was removed successfully.
+        let accounts = manager.get_accounts().await.unwrap();
+        assert!(accounts.len() == 1);
+        assert_eq!(
+            *accounts.get(0).unwrap().read().await.index(),
+            *first_account.read().await.index()
+        );
+
+        // Remove `first_account`.
+        let _ = manager
+            .remove_latest_account()
+            .await
+            .expect("cannot remove latest account");
+
+        // Check if the `first_account` was removed successfully. All accounts should be removed.
+        let accounts = manager.get_accounts().await.unwrap();
+        assert!(accounts.is_empty());
+
+        // Try remove another time (even if there is nothing to remove).
+        let _ = manager
+            .remove_latest_account()
+            .await
+            .expect("cannot remove latest account");
+
+        let accounts = manager.get_accounts().await.unwrap();
+        assert!(accounts.is_empty());
+
+        // Add two new accounts and return their index.
+
+        let third_account = manager.create_account().finish().await?;
+        let fourth_account = manager.create_account().finish().await?;
+
+        assert_eq!(manager.get_accounts().await.unwrap().len(), 2);
+
+        let third_account_index = *third_account.read().await.index();
+        let fourth_account_index = *fourth_account.read().await.index();
+
+        (third_account_index, fourth_account_index)
```

Comment:
The names are a little bit confusing, because the account indexes would be 0 and 1
Also it might be better to only recreate one account, so when we restore it from the db, one account is still removed, if re add both accounts before restoring it, the db might not be updated and the check wouldn't notice it, since we recreated the accounts already

Reply:
When recreating only 1 account, the test fails now. Not fully sure if I understand why? do you have an idea?

Reply:
Seems like that the database does not get updated in-time. But this shouldn't happen, right?